### DISCUSSION TITLE
使用set_prefixname更改输出动态库文件名后，怎么在其他引用库里使用add_deps?

### DISCUSSION THREAD

**Discussion by jiezenghuang** [xmake-io/xmake#4498] [General]
项目添加了nlk_前缀，io模块依赖common

target("io")
    add_headerfiles("*.h")
    add_files("*.cpp")
    add_deps("common")

checking for Microsoft Visual Studio (x64) version ... 2022
checking for Microsoft C/C++ Compiler (x64) version ... 19.38.33133
[ 100%]: linking.release nlk_common.dll
[ 90%]: linking.release nlk_io.dll
error: LINK : fatal error LNK1181: cannot open input file 'common.lib'

如果使用add_deps("nlk_common")，则提示找不到nlk_common模块

**Comment by waruqi** [comment]
你应该改 basename 而不是 prefixname , `set_basename("nlk_common")`

prefixname 改的是 libxxx.a 前面的 `lib` ，不管你改成什么，在 linux 下，都会影响 link 。。只不过恰好 win 上默认是 "" 空而已。。

**Comment by jiezenghuang** [comment]
set_basename是直接把target名称改了吧。不能统一加一个项目的前缀吗？类似boost，每个模块的dll文件名生成为：[package]_[component]_[mt]_[d].dll

用户在使用的时候还是可以用component名称去链接特定的库。

**Reply by jiezenghuang** [reply]
理解你意思了。有类似project name，target name这样的变量吗？帮助文档好像没找到。

**Reply by waruqi** [reply]
> set_basename是直接把target名称改了吧。

不是，只改文件名
